src/FixedDataTable.js

```diff
@@ -486,6 +486,7 @@ var FixedDataTable = React.createClass({
           scrollLeft={state.scrollX}
           fixedColumns={state.groupHeaderFixedColumns}
           scrollableColumns={state.groupHeaderScrollableColumns}
+          maxVisibleColumns={state.maxVisibleColumns}
```

Comment:
Right, I'm suggesting you could remove this whole calculation and passing down this variable, and instead just replace the iterations over maxVisibleColumns with iterations over newColumnsToRender or oldColumns respectively.  You'd need to add some logic for handling length mismatches, but I don't think it would be bad.

You could probably even keep the same code and replace props.maxVisibleColumns with max(newColumnsToRender.length, this._columnsToRender.length)